src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -336,12 +336,14 @@ private class FunctionActivityTracker : IDisposable
         {
             private readonly IMetricsPublisher _metricsPublisher;
             private readonly ILinuxContainerActivityPublisher _linuxContainerActivityPublisher;
+            private readonly object _runningFunctionsSyncLock = new object();
 
             private ulong _totalExecutionCount = 0;
+            private int _activeFunctionCount = 0;
             private int _functionActivityFlushInterval;
             private CancellationTokenSource _etwTaskCancellationSource = new CancellationTokenSource();
             private ConcurrentQueue<FunctionMetrics> _functionMetricsQueue = new ConcurrentQueue<FunctionMetrics>();
-            private ConcurrentDictionary<string, RunningFunctionInfo> _runningFunctions = new ConcurrentDictionary<string, RunningFunctionInfo>();
```

Comment:
Use of this concurrent dictionary was causing a "hot lock" to show up in profiles. I've moved to a simpler locking strategy.